File: java-sfn-unit-test/src/main/java/com/sfnut/sfn/wait/SfnWait.java
package com.sfnut.sfn.wait;

import com.sfnut.sfn.SfnStepWithPayload;
import com.sfnut.sfn.StepType;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.Duration;

public class SfnWait<PAYLOAD> extends SfnStepWithPayload<PAYLOAD> {
    @JsonProperty("StepName")
    private final String stepName;
    
    @JsonProperty("Duration")
    private final Duration duration;

    public SfnWait(String stepName, Duration duration) {
        super(StepType.WAIT);
        this.stepName = stepName;
        this.duration = duration;
    }

    @Override
    public PAYLOAD execute(PAYLOAD input) {
        // TODO: Increase fake time
        return input;
    }

    public String getStepName() {
        return stepName;
    }

    public Duration getDuration() {
        return duration;
    }
}

File: sfn/wait/SfnWait.cs
namespace sfn_ut.sfn.wait;

public class SfnWait<PAYLOAD>(string stepName, TimeSpan duration) 
  : SfnStepWithPayload<PAYLOAD>(StepType.Wait) {
    
  public readonly string StepName = stepName;
  public readonly TimeSpan Duration = duration;

  public override PAYLOAD Execute(PAYLOAD input) {
    // TODO: Increase fake time
    return input;
  }
}